ovs/lib/monitoring.py

```diff
@@ -71,14 +71,14 @@ def verify_vdisk_cache_quota():
         """
         MonitoringController._logger.info('Starting vDisk caching quota verification...')
         alba_guid_size_map = {}
+        vdisk_cache_quota_mapping = {'fragment_cache': 'fragment',
+                                     'block_cache': 'block'}
         for storagedriver in StorageDriverList.get_storagedrivers():
             storagedriver.invalidate_dynamics(['vpool_backend_info', 'vdisks_guids'])
-
-            for cache_type in [['cache_quota_fc', 'backend_info', 'connection_info', 'fragment'],
-                               ['cache_quota_bc', 'block_cache_backend_info', 'block_cache_connection_info', 'block']]:
-                cache_quota = storagedriver.vpool_backend_info[cache_type[0]]
-                backend_info = storagedriver.vpool_backend_info[cache_type[1]]
-                connection_info = storagedriver.vpool_backend_info[cache_type[2]]
+            for cache_type, cache_type_data in storagedriver.vpool_backend_info['caching_info'].iteritems():
+                cache_quota = cache_type_data['quota']
+                backend_info = cache_type_data['backend_info']
+                connection_info = cache_type_data.get('backend_info', {}).get('connection_info', None)
```

Comment:
`backend_info` is always filled out in the new structure, being `None` when there is no backend involved, resulting in an error when doing the `get`. Instead the `vpool_backend_info` should always return a dict for `backend_info` imo